### PR TITLE
create integration test framework, improve integration test

### PR DESCRIPTION
**Pull request by EvanBoyle**
I created a more robust integration test runner that attempt to handle errors a little better. I factored this out into a reusable framework under `/testing/integration`. I also added some logic to the actual integration test that issues a few queries to athena for each table that we create, and ensures that it get's some records back.

### CODE REVIEW

File: testing/integration/index.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        } catch (e) {
+            return Promise.resolve({ success: false, error: e });
+        }
+    }
+
+    public async teardown(): Promise<RunnerResult> {
+        try {
+            const destoryResult = await this.destroy();
+            if (!destoryResult.success) {
+                return Promise.resolve({ success: false, error: destoryResult.error })
```
Comment (jmaysrowland) [inline testing/integration/index.ts]: destroyResult not destoryResult

File: testing/integration/index.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    private async stackRemove(): Promise<CommandResult> {
+        return await execCommand("pulumi", ["stack", "rm", this.stackName, "--cwd", this.pulumiProjDir, "--yes"]);
+    }
+
+    private createConfigArray(config: { [key: string]: string }): string[] {
+        const result: string[] = [];
+        for (const key of Object.keys(config)) {
+            result.push("-c");
```
Comment (jmaysrowland) [inline testing/integration/index.ts]: Why the "-c" between every config line?

File: examples/datawarehouse/__tests__/streamingInputTable.integration.test.ts [hunk 3]
```diff
@@ -1,68 +1,76 @@
...
-        done()
+beforeAll(async () => {
+    const config: { [key: string]: string } = {
+        "aws:region": region,
+        "pulumi-serverless-db:shards": '{"prod": 8, "gamma": 4, "beta": 1}',
+        "pulumi-serverless-db:stage": "beta",
+        "pulumi-serverless-db:dev": "true"
     };
 
+    const pulumiProjDir = resolve("./examples/datawarehouse");
```
Comment (jmaysrowland) [inline examples/datawarehouse/__tests__/streamingInputTable.integration.test.ts]: Should this test have it's own project that only creates what is needed for the test?
Definitely something that can be done later, but just a thought.

File: testing/integration/index.ts (hunk 2 above)
Comment (EvanBoyle) [reply testing/integration/index.ts]: As far as I can tell, there's no way to pass all of these values under one flag. For each config value you specify you have to use `-c`: `-c foo=bar -c fizz=buzz ...`

File: examples/datawarehouse/__tests__/streamingInputTable.integration.test.ts (hunk 3 above)
Comment (EvanBoyle) [reply examples/datawarehouse/__tests__/streamingInputTable.integration.test.ts]: I agree with this. Right now this is sort of an integration test on the example. Likely we'll need to fork off at some point.

**Comment by EvanBoyle** [comment]
fixes #9

**Pull request merged**